satpy/readers/lmi_l2.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Lightning Mapping Imager Dataset reader
+
+The files read by this reader are described in:
+
+    http://fy4.nsmc.org.cn/data/en/data/realtime.html
+
+"""
+
+from satpy.readers.netcdf_utils import NetCDF4FileHandler, netCDF4
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class LMIL2FileHandler(NetCDF4FileHandler):
+    """File handler for LMI L2 netCDF files."""
+
+    @property
+    def start_time(self):
+        return self.filename_info['start_time']
+
+    @property
+    def end_time(self):
+        return self.filename_info['end_time']
+
+    @property
+    def platform_shortname(self):
+        return self.filename_info['platform_id']
+
+    @property
+    def sensor(self):
+        return self.filename_info['instrument'].lower()
+
+    def available_datasets(self, configured_datasets=None):
+        """Automatically determine datasets provided by this file."""
+        logger.debug("Available_datasets begin...")
+
+        # Determine shape of the geolocation data (lat/lon)
+        lat_shape = None
+        for var_name, _val in self.file_content.items():
+            # Could probably avoid this hardcoding, will think on it
```

Comment:
If you know there is only one latitude dataset then you could load only the `LAT` *or* you could look through the configured datasets for one with `standard_name: latitude`. Just an idea.

Reply:
Embed this into the part of reading configured datasets

Reply:
@djhoese I tested this again and found a bug: some data like this `{'name': 'EOT', 'file_key': 'EOT', 'file_type': 'lmi', 'resolution': '7800m'}` don't have the `standard_name` key in the `ds_info` dict, because we don't add it to the `yaml` file. So, I switched to use `name` for this purpose.